### PR TITLE
Introduce ethsign msg, allowing signing of arbitrary data

### PR DESCRIPTION
**Pull request by nanexcool**
This is a big one :)

Primarily, it adds a `ethsign message` or `ethsign msg` command, which allows signing arbitrary data (in hex) similarly to the RPC call `eth_sign`. (see https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_sign ) It hashes whatever data the use inputs along with a known header to prevent accidental signing of transactions. This is the standard behavior in Geth and Parity.

This is very useful for interacting with smart contracts that make use of `ecrecover` such as the new price feeds.

My Golang plugin also formatted some of the code according to best practices (sorry about that). I can try to put it back as it was.

Also, refactored some code into a function, could be refactored further, especially the part where you get the wallet and wait for a passphrase or ledger/trezor.

### CODE REVIEW

**Comment by mbrock** [comment]
This is amazing! 🕺 I'll look a bit more closely after breakfast.

I guess it would be somewhat more convenient to have the code reformatting as a separate commit, but it's such a small program that it doesn't matter too much. Using the standard Go formatting is a good idea.

The reason I hadn't done that refactoring already is that I was literally too lazy to look up like how to even make functions in Go or something. 😬

File: ethsign.go [hunk 1]
```diff
@@ -296,19 +287,141 @@ func main() {
...
+				data := hexutil.MustDecode(dataString)
+
+				wallets := getWallets(c.StringSlice("key-store"), defaultKeyStores)
+
+				var wallet accounts.Wallet
+				var acct *accounts.Account
+
+				needPassphrase := true
+
+			Scan:
```
Comment (mbrock) [inline ethsign.go:353]: I suppose this loop and also the following passphrase stuff could also be refactored to a new function?

**Comment by nanexcool** [comment]
As you're having breakfast I'm heading to bed, so I can definitely split the commits in two, but it'll be in a few hours. ;) I think it is indeed a better idea. (I use Visual Studio Code, installed some recommended Go extensions and all of a sudden it started formatting everything)

Also, on the same boat regarding Go functions. Managed to refactor one tiny part, but hit a wall when tackling the more complex parts!

File: ethsign.go (hunk 1 above)
Comment (nanexcool) [reply ethsign.go:353]: Yeah, it's really the command inputs and eventual signing that changes between `tx` and `msg`. Everything in between is the same.

**Comment by nanexcool** [comment]
Closing this, will submit individual PRs

**Pull request closed without merging**